### PR TITLE
Update golang to 1.13.0

### PR DESCRIPTION
**Pull request by tao12345666333**
replcace #39

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,7 +22,9 @@ ENV OSX_CROSS_PATH=/osxcross
 FROM base AS osx-sdk
 ARG OSX_SDK
 ARG OSX_SDK_SUM
-ADD https://s3.dockerproject.org/darwin/v2/${OSX_SDK}.tar.xz "${OSX_CROSS_PATH}/tarballs/${OSX_SDK}.tar.xz"
+# ADD https://s3.dockerproject.org/darwin/v2/${OSX_SDK}.tar.xz "${OSX_CROSS_PATH}/tarballs/${OSX_SDK}.tar.xz"
+# FIXME: upload to s3.dockerproject.org
+ADD https://github.com/ndeloof/golang-cross/raw/113fix/${OSX_SDK}.tar.xz "${OSX_CROSS_PATH}/tarballs/${OSX_SDK}.tar.xz"
```
Comment (tao12345666333) [inline Dockerfile]: I used the SDK https://github.com/phracker/MacOSX-SDKs, but unfortunately it lacks the necessary headers file. 
I got some error like this:

```
testing o32-clang ... osxcross: error: cannot find libc++ headers
osxcross: error: while detecting target
```
The full job can be found here https://gitlab.com/taobeier/golang-cross/-/jobs/308408417

Later I chose the SDK build by @ndeloof  It works well. https://gitlab.com/taobeier/golang-cross/-/jobs/309123113

I think we shoud upload new version SDK to s3.dockerproject.org @thaJeztah

File: Dockerfile [hunk 2]
```diff
@@ -1,17 +1,17 @@
-ARG GO_VERSION=1.12.10
+ARG GO_VERSION=1.13.0
 
 # OS-X SDK parameters
 # NOTE: when changing version here, make sure to also change OSX_CODENAME below to match
-ARG OSX_SDK=MacOSX10.10.sdk
-ARG OSX_SDK_SUM=631b4144c6bf75bf7a4d480d685a9b5bda10ee8d03dbf0db829391e2ef858789
+ARG OSX_SDK=MacOSX10.11.sdk
+ARG OSX_SDK_SUM=98cdd56e0f6c1f9e1af25e11dd93d2e7d306a4aa50430a2bc6bc083ac67efbb8
```
Comment (tao12345666333) [inline Dockerfile]: Once the SDK uploads to s3.dockerproject.org, I can change here. (If you want to re-generate it)

**Comment by tao12345666333** [comment]
> I have no plan to keep this sdk file hosted there, it's just there for demonstration purpose.

Yep. We will upload it to s3.dockerproject . （Just because I didn't rebuild it :smile:

**Comment by tao12345666333** [comment]
```
Successfully tagged docker-cli-cross:latest
docker run --rm -e VERSION=19.09.0-dev -e GITCOMMIT -e PLATFORM -e TESTFLAGS -e TESTDIRS -e GOOS -e GOARCH -e GOARM -v "/builds/taobeier/golang-cross/cli":/go/src/github.com/docker/cli -v "docker-cli-dev-cache:/root/.cache/go-build"  docker-cli-cross make cross
./scripts/build/cross
Building binaries for all platforms
Generating windows resources
Building build/docker-windows-amd64
/go/src/github.com/docker/cli/scripts/gen/../winresources/common.rc:11: digit exceeds base
/go/src/github.com/docker/cli/scripts/gen/../winresources/common.rc:12: digit exceeds base
/go/src/github.com/docker/cli/scripts/gen/../winresources/common.rc:11: digit exceeds base
/go/src/github.com/docker/cli/scripts/gen/../winresources/common.rc:12: digit exceeds base
Building build/docker-darwin-amd64
Building statically linked build/docker-linux-amd64
Building statically linked build/docker-linux-arm
Building statically linked build/docker-linux-ppc64le
Building statically linked build/docker-linux-s390x
```

Using GitLab CI test the `cli`'s build job. https://gitlab.com/taobeier/golang-cross/-/jobs/309123114

**Comment by ndeloof** [comment]
IIUC from a legal PoV we are not allowed to host sdk (derived from xcode) on a public s3, which can be considered a form of "redistribution".

**Comment by tao12345666333** [comment]
I have not studied this point. I noticed that the previous sdk was hosted on S3. 
I also saw a similar project https://github.com/phracker/MacOSX-SDKs but I am not familiar with it.

File: Dockerfile [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 
-# OSX-cross parameters. Go 1.11 requires OSX >= 10.10
-ARG OSX_VERSION_MIN=10.10
+# OSX-cross parameters. Go 1.13 requires OSX >= 10.11
+ARG OSX_VERSION_MIN=10.11
 ARG OSX_CROSS_COMMIT=a9317c18a3a457ca0a657f08cc4d0d43c6cf8953
```
Comment (tao12345666333) [inline Dockerfile]: If we need to update to master I have already done it. https://github.com/tao12345666333/golang-cross/commit/afc6f85b4329acb8bb2f906a6b082f0bd840fb8b  

The CI has been passed. https://gitlab.com/taobeier/golang-cross/-/jobs/309376124

**Comment by tao12345666333** [comment]
@thaJeztah  PTAL. Thanks.

**Comment by tao12345666333** [comment]
ping @thaJeztah  PTAL. Thanks. 

cc @kolyshkin

**Comment by thaJeztah** [comment]
Still trying to get an answer on updating the s3 bucket; hold tight 😅 
Probably we should update to Go 1.13.3 now as well 

ping @andrewhsu PTAL

**Comment by pchampio** [comment]
Hey, What's the status of this PR?
Awesome job btw!

**Comment by tao12345666333** [comment]
wait for s3 bucket.

**Comment by pchampio** [comment]
What is the s3 bucket used for?
how long will it takes?
Thanks for your answers!

**Comment by Jacalz** [comment]
Could we please get go updated to 1.13.5 and see this merged? I can not use this package to cross compile my application as it relies on `os.UserConfigDir()` that was added in version 1.13...

**Comment by tao12345666333** [comment]
> Could we please get go updated to 1.13.5 and see this merged?

Sorry, unfortunately, if we want to merge this PR, we need a bucket to hold the SDK. 

As @thaJeztah  commented before https://github.com/docker/golang-cross/pull/45#issuecomment-545827442


> Still trying to get an answer on updating the s3 bucket; hold tight

**Comment by tao12345666333** [comment]
Update for solve conflict file.

**Comment by Jacalz** [comment]
It is sad that this seems to take such a long time and that the code needs to workaround using functions introduced in 1.13.0. I would be great to get this finished 👍 

Anyhow, while we already are waiting, could we please get the go version updated to 1.13.6 so we get all the latest fixes and security patches? 🙂

**Comment by tao12345666333** [comment]
@Jacalz  The reason this PR has never been merged is because we don't have hostable S3 storage.
https://github.com/docker/golang-cross/pull/45#issuecomment-545827442

I can update the Go version and it ’s easy.

**Comment by geek1011** [comment]
Note that there are other images similar to this one, for example https://github.com/elastic/golang-crossbuild.

**Comment by robdefeo** [comment]
Any update on this?

**Comment by robdefeo** [comment]
@thaJeztah https://github.com/phracker/MacOSX-SDKs/releases would this not provide a drop in replacement for S3 file hosting?

**Comment by tao12345666333** [comment]
@robdefeo Thanks! 
But I have tried it before, it is incomplete and cannot be compiled for us.

**Comment by robdefeo** [comment]
@tao12345666333 what are the exact requirements needed?  This will help the community figure know exactly what is needed then we will be more easily able to help. Thanks

**Comment by tao12345666333** [comment]
@robdefeo  ref: https://github.com/phracker/MacOSX-SDKs/issues/21

**Comment by Jacalz** [comment]
What exactly is the problem with the bucket for storing the SDK?
Is it impossible to do or have nobody stepped up with a working hosting bucket?

**Comment by thaJeztah** [comment]
No approval from legal to upload a new version of the SDK.

I had an alternative stashed in my local Git, that just updates the Go version, and adds some TODO comments. One restriction of that change would be that the image wouldn't set the correct "minimum version"; perhaps it doesn't cause issues, but added the comments for future follow up.

Let me open a PR for that (at least as a temporary solution)

**Comment by thaJeztah** [comment]
GitHub is giving angry unicorns, but managed to push https://github.com/docker/golang-cross/pull/53

**Pull request closed without merging**